src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -167,17 +167,7 @@ function BaseSelectionList<TItem extends User | RadioItem>(
             const itemIndex = item.index ?? -1;
             const sectionIndex = item.sectionIndex ?? -1;
 
-            // Note: react-native's SectionList automatically strips out any empty sections.
-            // So we need to reduce the sectionIndex to remove any empty sections in front of the one we're trying to scroll to.
-            // Otherwise, it will cause an index-out-of-bounds error and crash the app.
-            let adjustedSectionIndex = sectionIndex;
-            for (let i = 0; i < sectionIndex; i++) {
-                if (sections[i].data) {
-                    adjustedSectionIndex--;
-                }
-            }
-
-            listRef.current.scrollToLocation({sectionIndex: adjustedSectionIndex, itemIndex, animated, viewOffset: variables.contentHeaderHeight});
+            listRef.current.scrollToLocation({sectionIndex, itemIndex, animated, viewOffset: variables.contentHeaderHeight});
```

Comment:
Even though the issue happens on OptionsSelector, SelectionList applies the same logic, so I removed it here too. Here is a video showing the scroll still works fine.

https://github.com/Expensify/App/assets/50919443/ffe16303-2ec5-4ffa-ba58-f030173a3dc9